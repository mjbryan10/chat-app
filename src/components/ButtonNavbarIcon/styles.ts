import styled, { ThemeProps, Theme } from 'styled-components';
import { Props } from '.';

interface ButtonProps extends ThemeProps<Theme>, Props {}

export const Button = styled.button<ButtonProps>`
   box-sizing: border-box;
   height: ${({ height }) => (height || 50)}px;
   width: ${({ width }) => (width || 50)}px;
   color: ${({ color, theme }) =>
      color ? theme.color[color] : theme.color.text || 'inherit'};
   font-size: ${({height}) => height ? height*0.8 : 50*0.8}px;
   background: none;
   border: none;
   margin: 0 5px;
   &:hover {
      cursor: pointer;
   }
`;

//TODO: Consider removing margin;