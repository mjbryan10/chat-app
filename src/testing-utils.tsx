import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }