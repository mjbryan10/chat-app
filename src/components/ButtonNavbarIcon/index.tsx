import React, { FC, SyntheticEvent } from 'react';
import * as S from './styles';
import { ThemeColor } from 'shared/theme/types';

export interface Props {
   height?: number;
   width?: number;
   color?: ThemeColor;
   handleClick?: () => void;
}

const ButtonNavbarIcon: FC<Props> = ({ children, height, width, color, handleClick }) => {

   const onClick = (event: SyntheticEvent<HTMLButtonElement>) => {
      event.preventDefault();
      if (handleClick) handleClick();
   };
   return (
      <S.Button height={height} width={width} color={color} onClick={onClick}>
         {children}
      </S.Button>
   );
};

export default ButtonNavbarIcon;
