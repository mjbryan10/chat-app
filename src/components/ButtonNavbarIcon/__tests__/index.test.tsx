import React from 'react';
import { render } from 'testing-utils';
import ButtonNavbarIcon from '..';

describe('ButtonNavbarIcon', () => {
  beforeEach(() => {
  })
  test('should display the given children as its value', () => {
    const testString = "Test";
    const utils = render(<ButtonNavbarIcon>{testString}</ButtonNavbarIcon>)
    const {getByText} = utils;

    expect(getByText(/test/i)).toBeVisible();

  })
  
})
