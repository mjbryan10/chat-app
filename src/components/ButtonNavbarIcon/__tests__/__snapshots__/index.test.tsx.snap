// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonNavbarIcon should render correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm iaTDms"
    data-testid="button-navbar-icon"
  />
</DocumentFragment>
`;
